### DISCUSSION TITLE
Can not  dateFormatter give YearMonth.class

### DISCUSSION THREAD

**Discussion by nqt-tech** [ozlerhakan/poiji#254] [Q&A] [answered]
Currently I only see dateFormatter give Date.class, LocalDate.class and LocalDateTime.class. Is there any way I can format the YearMonth.class?

**Comment by ozlerhakan** [comment, marked as answer]
Hi @nqt-tech ,

Currently the YearMonth.class is not supported in Poiji. Do you want to contribute to the project? Another option is you can create your  custom Config so that you can cast the value to the YearMonth.class :)

**Comment by ozlerhakan** [comment]
Here is an example of how to do :
```
     public class MyCustomConfig implements Casting {
        @Override
        public Object castValue(Field fieldType, String value, int row, int column, PoijiOptions options) {
            Class<?> field = fieldType.getType();
            if (field == YearMonth.class) {
                // do something
            }
            return value.trim();
        }
    }

PoijiOptions poijiOptions = PoijiOptions.PoijiOptionsBuilder.settings()
                .withCasting(new MyCustomConfig())
                .build();
```
Let me known if It helps.

**Reply by nqt-tech** [reply]
thank you @ozlerhakan for reply, at getValueObject() method missing YearMonth.class so the example you give will be error. I have tried with that before. But it's okay I already have another way of using annotation to validate, but I still want poiji to support it, because it's basically the same as dateFormatter LocalDateTime.class

**Reply by ozlerhakan** [reply]
ah I see; you can accept the question as resolved and create an issue, so that we can takr a look at it. Thanks @nqt-tech !
